phoenicis-engines/src/main/java/org/phoenicis/engines/EngineSettingsManager.java

```diff
@@ -78,27 +87,30 @@ public void fetchAvailableEngineSettings(RepositoryDTO repositoryDTO,
 
     /**
      * retrieves a Javascript string which can be used to fetch the available settings
+     *
      * @param repositoryDTO repository containing the settings
      * @return Javascript
      */
     private String createFetchScript(RepositoryDTO repositoryDTO) {
```

Comment:
@plata is there a special reason why we fetch the settings through this script? 

I think the returned script is really difficult to understand because it is generated on the fly. A much better solution in my eyes would be to load the scripts individually via the `getSetting` method above. We could do this as a kind of batch job in which all settings for all supported engines are returned, like we do currently, or we load the settings on demand whenever a container has been opened.

@plata @qparis what do you think?

Reply:
When I created this, I could not manage to get all settings directly. If you have an idea how this could work, I would be happy to remove this method. However, I don't think that this should be part of this pull request.